code/game/objects/items/weapons/grenades/grenade.dm

```diff
@@ -115,3 +114,7 @@
 	///We need to clear the walk_to on destroy to allow a grenade which uses walk_to or related to properly GC
 	walk_to(src, 0)
 	return ..()
+
+/obj/item/grenade/cmag_act(mob/user)
+	to_chat(user, "<span class='warning'>You drip some yellow ooze into [src]. [src] suddenly doesn't want to leave you...</span>")
+	AddComponent(/datum/component/boomerang, throw_range, TRUE)
```

Comment:
is there a procedure for un cmagging a grenade?

Reply:
Not yet, I'll add it!